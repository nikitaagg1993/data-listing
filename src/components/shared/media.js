import { css } from 'styled-components';

export const SMALL = 599;
export const MEDIUM = 959;
export const LARGE = 1280;

export const media = {
  sm: (...args) => css`
    @media screen and (max-width: ${SMALL}px) {
      ${css(...args)}
    }
  `,
  md: (...args) => css`
    @media screen and (max-width: ${MEDIUM}px) {
      ${css(...args)}
    }
  `,
  lg: (...args) => css`
    @media screen and (max-width: ${LARGE}px) {
      ${css(...args)}
    }
  `,
  custom: (value) => (...args) => css`
    @media screen and (max-width: ${value}px) {
      ${css(...args)}
    }
  `,
};
