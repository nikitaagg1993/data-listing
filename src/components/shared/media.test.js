import { media } from './media';

describe('media', () => {
  test('media: sm', () => {
    expect(media.sm([])).toMatchSnapshot();
  });
  test('media: md', () => {
    expect(media.md([])).toMatchSnapshot();
  });
  test('media: lg', () => {
    expect(media.lg([])).toMatchSnapshot();
  });
  test('media: custom', () => {
    expect(media.custom('abc')('display: none')).toMatchSnapshot();
  });
});
