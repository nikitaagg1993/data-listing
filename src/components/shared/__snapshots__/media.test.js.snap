// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`media media: custom 1`] = `
Array [
  "
    @media screen and (max-width: ",
  "abc",
  "px) {
      ",
  "d",
  "
    }
  ",
]
`;

exports[`media media: lg 1`] = `
Array [
  "
    @media screen and (max-width: ",
  "1280",
  "px) {
      ",
  "
    }
  ",
]
`;

exports[`media media: md 1`] = `
Array [
  "
    @media screen and (max-width: ",
  "959",
  "px) {
      ",
  "
    }
  ",
]
`;

exports[`media media: sm 1`] = `
Array [
  "
    @media screen and (max-width: ",
  "599",
  "px) {
      ",
  "
    }
  ",
]
`;
